EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigi.cc

```diff
@@ -314,17 +316,32 @@ void SiPixelRawToDigi::produce( edm::Event& ev,
     hCPU->Fill( theTimer->realTime() ); 
     hDigi->Fill(formatter.nDigis());
   }
+  
 
+  //------------------------------------
   //send digis and errors back to framework 
+
+  edm::Handle<PixelFEDChannelCollection> pixelFEDChannelCollectionHandle;
+  std::unique_ptr<PixelFEDChannelCollection> PixelFEDChannelCollection_ = nullptr;  
+  if (ev.getByToken(theBadPixelFEDChannelsLabel, pixelFEDChannelCollectionHandle)){
```

Comment:
@makortel could you please check the comment below?

Reply:
@markotel I would argue that creating an enable/disable would itself be a source of potential mis-configuration. 1) when run on data, the input cannot exist, so enabling the feature would throw an unnecessary exception for no good reason, 2) when run on simulation, disabling while the digitizer is configured to simulate bad component would not be a good idea, but would pass unnoticed following your suggestion.

Reply:
Good point. Is the plan then eventually to pack this information in the RAW for MC as well?

Reply:
Yes, we were planning to do so in order to avoid the kind of ambiguity you pointed out, otherwise we might end up having to configure RawToDigi separately for MC; although, one might argue that packing this info into RAW is an unjustified complication at the same time.

Reply:
But it's in the RAW for data, right? I suppose the same arguments hold here as for packing anything to RAW for MC.

Reply:
Agreed, reluctantly. This is partly why we suggested to factorize this development into two steps.

Reply:
Ok, then I agree that this is likely the best compromise for this PR. Thanks for the clarifications.